sisl/viz/plotly/plots/bands.py

```diff
@@ -865,3 +888,67 @@ def _plot_Kdiff(self, band1, band2, E=None, offsetE=False):
         plt.update_layout({"title": f"Delta K between bands {band1} and {band2}", 'xaxis_title': 'Delta k', 'yaxis_title': 'Energy [eV]'})
 
         return plt
+
+    def effective_mass(self, band, k, band_spin=0, n_points=10, k_direction="symmetric"):
+        """
+        Calculates the effective mass from the curvature of a band in a given k point.
+
+        NOTE: The calculations done here are only valid if there are no band crossings in the
+        range that is being fitted.
+
+        Parameters
+        -----------
+        band: int
+            The index of the band that we want to fit
+        k: float or str
+            The k value where we want to find the curvature of the band to calculate the effective mass.
+        band_spin: int, optional
+            The spin value for which we want the effective mass.
+        n_points: int
+            The number of points that we want to use for the polynomial fit.
+        k_direction: {"symmetric", "right", "left"}, optional
+            Indicates in which direction -starting from `k`- should the band be fitted. 
+            "left" and "right" mean that the fit will only be done in one direction, while
+            "symmetric" indicates that points from both sides will be used.
+        
+        Return
+        -----------
+        float
+            The efective mass, in atomic units.
+        """
+        from sisl.unit.base import units
+
+        # Get the band that we want to fit
+        band_vals = self.bands.sel(band=band, spin=band_spin)
+
+        # Sanitize k to a float
+        k = self._sanitize_k(k)
+        # Find the index of the requested k
+        k_index = abs(self.bands.k -k).values.argmin()
+
+        # Determine which slice of the band will we take depending on k_direction and n_points
+        if k_direction == "symmetric":
+            sel_slice = slice(k_index - n_points // 2, k_index + n_points // 2 + 1)
+        elif k_direction == "left":
+            sel_slice = slice(k_index - n_points + 1, k_index + 1)
+        elif k_direction == "right":
+            sel_slice = slice(k_index, k_index + n_points)
+        else:
+            raise ValueError(f"k_direction must be one of ['symmetric', 'left', 'right'], {k_direction} was passed")
+        
+        # Grab the slice of the band that we are going to fit
+        sel_band = band_vals[sel_slice] * units("eV", "Hartree")
```

Comment:
Should there be some check that the slice actually has a length of 4 or some minimum length? I.e. what if users have too few points?

Reply:
Numpy will already raise an error (warning) which is quite clear if not enough (very few) points are passed to `np.polyfit`.

Reply:
Ok, fair enough ;)